.github/workflows/docker.yml

```diff
@@ -0,0 +1,66 @@
+name: Build & Push Docker Images
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  pre_job:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+            fetch-depth: 0
+
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+            username: ${{ secrets.DOCKERHUB_USERNAME }}
+            password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: List built images
+        run: docker images
+
+      - name: Build and Push Docker Image (Without Solvers)
+        env:
+          IMAGE_NAME: pybamm/pybamm:latest
+        run: |
+          docker build -t $IMAGE_NAME -f scripts/Dockerfile .
+          docker push $IMAGE_NAME
```

Comment:
We would need to use the official Docker action to push images (`docker/build-push-action`) and point it to a path where the built images eill be stored

Reply:
Thanks, will look into it.